embedding/embedlite/embedshared/nsWindow.cpp

```diff
@@ -276,11 +285,57 @@ layers::LayersId nsWindow::GetRootLayerId() const
   return mCompositorSession ? mCompositorSession->RootLayerTreeId() : layers::LayersId{0};
 }
 
-void nsWindow::SetContentController(mozilla::layers::GeckoContentController *aController)
+#ifdef MOZ_LOGGING
+static void LogControllerStack(std::list<EmbedContentController *> &controllers)
+{
+  int pos = 0;
+  for (std::list<EmbedContentController *>::const_iterator it = controllers.begin(); it != controllers.end(); ++it) {
+    LOGT("Stack controller %d : %u", pos, (*it)->GetUniqueID());
+    ++pos;
+  }
+}
+#endif // MOZ_LOGGING
+
+void nsWindow::SetContentController(EmbedContentController *aController)
 {
   if (mCompositorSession) {
     mCompositorSession->SetContentController(aController);
   }
+
+  MOZ_ASSERT(aController);
+
+  // If the view is deactivated we need to reactivate the previous view's
+  // content controller, so we store the controllwers on a stack
+  if (mControllers.empty() || aController != mControllers.back()) {
+    LOGT("Pushing content controller %u", aController->GetUniqueID());
+    mControllers.push_back(aController);
+    LOG_CONTROLLER_STACK;
+  }
+}
+
+void nsWindow::DropContentController(EmbedContentController* aController)
```

Comment:
Wouldn't it be sufficient to have this called like ```nsWindow::Release(EmbedContentController *aController)```, would it? Maybe this could even be Release & Reserve pair. Somehow I find this repetition of ContentController unnecessary.

And yes like ```(EmbedContentController *aController)``` => there's a mixture of different styles in the embedlite we should be heading towards this style.